org.eclipse.lsp4e/src/org/eclipse/lsp4e/outline/OutlineViewHideSymbolKindMenuContributor.java

```diff
@@ -0,0 +1,87 @@
+/*******************************************************************************
+ * Copyright (c) 2024 Advantest Europe GmbH. All rights reserved.
+ * This program and the accompanying materials are made
+ * available under the terms of the Eclipse Public License 2.0
+ * which is available at https://www.eclipse.org/legal/epl-2.0/
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ *
+ * Contributors:
+ *  Dietrich Travkin (Solunar GmbH) - initial implementation of outline contents filtering (issue #254)
+ *******************************************************************************/
+package org.eclipse.lsp4e.outline;
+
+import java.util.Arrays;
+import java.util.Comparator;
+
+import org.eclipse.core.runtime.preferences.IEclipsePreferences;
+import org.eclipse.core.runtime.preferences.InstanceScope;
+import org.eclipse.jface.action.Action;
+import org.eclipse.jface.action.ActionContributionItem;
+import org.eclipse.jface.action.IAction;
+import org.eclipse.jface.action.IContributionItem;
+import org.eclipse.jface.resource.ImageDescriptor;
+import org.eclipse.lsp4e.LanguageServerPlugin;
+import org.eclipse.lsp4e.ui.LSPImages;
+import org.eclipse.lsp4j.SymbolKind;
+import org.eclipse.swt.graphics.Image;
+import org.eclipse.ui.actions.CompoundContributionItem;
+
+public class OutlineViewHideSymbolKindMenuContributor extends CompoundContributionItem {
+
+	@Override
+	protected IContributionItem[] getContributionItems() {
+		return Arrays.stream(SymbolKind.values())
+			.sorted(new Comparator<SymbolKind>() {
+
+				@Override
+				public int compare(SymbolKind sk1, SymbolKind sk2) {
+					return sk1.name().compareToIgnoreCase(sk2.name());
```

Comment:
```suggestion
					return sk1.name().compareTo(sk2.name());
```
should be enough since the names in the Enum will be constant